### PR TITLE
Prevent to write an old governance index again

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

If `init` is executed on the existing database, it may corrupt governance index data appending `0` in the governance index slice (The slice should have ascending order). With this PR, appending the same or smaller block number is not allowed on the governance index slice when it is stored on the database. 

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by KimKyungup — DISMISSED** [review]

File: storage/database/db_manager_test.go [hunk 1]
```diff
@@ -908,6 +909,80 @@ func TestDBManager_StateMigrationDBPath(t *testing.T) {
...
+			assert.Nil(t, err)
+			assert.Nil(t, dbm.GetMiscDB().Put(governanceHistoryKey, data))
+
+			// read and check the idnexes from the database
+			idxes, err = dbm.ReadRecentGovernanceIdx(0)
+			assert.Equal(t, testIdxes, idxes)
+			assert.Nil(t, err)
+		}
+
+		// unexpected case: the governance indexes in the database is not in ascending order
```
Comment (ehnuje) [inline storage/database/db_manager_test.go:965]: This case shouldn't be happened, right?

File: storage/database/db_manager.go [hunk 2]
```diff
@@ -2038,6 +2054,11 @@ func (dbm *databaseManager) ReadRecentGovernanceIdx(count int) ([]uint64, error)
 			return nil, e
 		}
 
+		// Make sure idxHistory should be in ascending order
+		sort.Slice(idxHistory, func(i, j int) bool {
+			return idxHistory[i] < idxHistory[j]
+		})
```
Comment (ehnuje) [inline storage/database/db_manager.go:2060]: If sort works here, that means the code in line#2016~line#2020 didn't work.
Is this code needed for handling (already) inserted values?

File: storage/database/db_manager_test.go (hunk 1 above)
Comment (aidan-kwon) [reply storage/database/db_manager_test.go:965]: Right. It shouldn't

File: storage/database/db_manager.go (hunk 2 above)
Comment (aidan-kwon) [reply storage/database/db_manager.go:2060]: @ehnuje You are right. `idxHistory` should be sorted already in the database. It is a kind of defensive programming to make sure the order. If `idxHistory` is corrupted, the Klaytn node could experience a bad block and hard to failback again. Indeed, the function will be called frequently. 
If you still think the code is not necessary, I will consider removing it.

Comment (ehnuje) [reply storage/database/db_manager.go:2060]: @aidan-kwon I think it's okay. However, why does this bug happen on KES environment?

File: storage/database/db_manager_test.go [hunk 3]
```diff
@@ -908,6 +909,80 @@ func TestDBManager_StateMigrationDBPath(t *testing.T) {
...
+		assert.NotNil(t, err)
+
+		// normal case
+		{
+			// write indexes on the database
+			data, err := json.Marshal(testIdxes)
+			assert.Nil(t, err)
+			assert.Nil(t, dbm.GetMiscDB().Put(governanceHistoryKey, data))
+
+			// read and check the idnexes from the database
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go:959]:
```suggestion
			// read and check the indexes from the database
```

File: storage/database/db_manager.go (hunk 2 above)
Comment (aidan-kwon) [reply storage/database/db_manager.go:2060]: @ehnuje 
1. When `ken init` is executed, `WriteGovernance` is called **without loading of index cache** `governance.idxCache` and `WriteGovernance` finally calls `WriteGovernanceIdx`.

https://github.com/klaytn/klaytn/blob/3281053dcbfab19821f32dbfcfdf03b1a9dc3d34/cmd/utils/nodecmd/chaincmd.go#L153-L158

2. `WriteGovernanceIdx` doesn't filter an index that is not in ascending order. So, I added the following logic inside of 
`WriteGovernanceIdx` also as `WriteGovernance` do. 
https://github.com/klaytn/klaytn/blob/3281053dcbfab19821f32dbfcfdf03b1a9dc3d34/storage/database/db_manager.go#L2016-L2020

File: storage/database/db_manager_test.go (hunk 3 above)
Comment (aidan-kwon) [reply storage/database/db_manager_test.go:959]: Thanks!

File: storage/database/db_manager_test.go [hunk 4]
```diff
@@ -908,6 +909,80 @@ func TestDBManager_StateMigrationDBPath(t *testing.T) {
...
+		{
+			invalidTestIdxes := append(testIdxes, testIdxes[0])
+			expectedIdxes := append([]uint64{testIdxes[0]}, testIdxes...)
+
+			// write invalid indexes on the database
+			data, err := json.Marshal(invalidTestIdxes)
+			assert.Nil(t, err)
+			assert.Nil(t, dbm.GetMiscDB().Put(governanceHistoryKey, data))
+
+			// read and check the idnexes from the database
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go:975]:
```suggestion
			// read and check the indexes from the database
```

**Review by ehnuje — APPROVED** [review]
LGTM except for @KimKyungup 's comment.

File: storage/database/db_manager_test.go (hunk 4 above)
Comment (aidan-kwon) [reply storage/database/db_manager_test.go]: Resolved!

**Review by KimKyungup — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by winnie-byun — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Pull request merged**